server/src/main/kotlin/org/javacs/kt/Compiler.kt

```diff
@@ -331,3 +344,49 @@ private fun describeExpression(expression: String): String = expression.lines().
         (lines.take(3) + listOf("...", lines.last())).joinToString(separator = "\n")
     }
 }
+
+class LanguageVersionSettingsImpl @JvmOverloads constructor(
+    override val languageVersion: LanguageVersion,
+    override val apiVersion: ApiVersion,
+    analysisFlags: Map<AnalysisFlag<*>, Any?> = emptyMap(),
+    specificFeatures: Map<LanguageFeature, LanguageFeature.State> = emptyMap()
+) : LanguageVersionSettings {
+    private val analysisFlags: Map<AnalysisFlag<*>, *> = Collections.unmodifiableMap(analysisFlags)
+    private val specificFeatures: Map<LanguageFeature, LanguageFeature.State> = Collections.unmodifiableMap(specificFeatures)
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T> getFlag(flag: AnalysisFlag<T>): T = analysisFlags[flag] as T? ?: flag.defaultValue
+
+    override fun getFeatureSupport(feature: LanguageFeature): LanguageFeature.State {
+        specificFeatures[feature]?.let { return it }
+
+        val since = feature.sinceVersion
+        if (since != null && languageVersion >= since && apiVersion >= feature.sinceApiVersion) {
+            return feature.defaultState
+        }
+
+        return LanguageFeature.State.DISABLED
+    }
+
+    override fun toString() = buildString {
+        append("Language = $languageVersion, API = $apiVersion")
+        specificFeatures.forEach { (feature, state) ->
+            val char = when (state) {
+                LanguageFeature.State.ENABLED -> '+'
+                LanguageFeature.State.ENABLED_WITH_WARNING -> '~'
+                LanguageFeature.State.ENABLED_WITH_ERROR, LanguageFeature.State.DISABLED -> '-'
+            }
+            append(" $char$feature")
+        }
+    }
+
+    override fun isPreRelease(): Boolean = languageVersion.isPreRelease() ||
+        specificFeatures.any { (feature, state) ->
+            state == LanguageFeature.State.ENABLED && feature.forcesPreReleaseBinariesIfEnabled()
+        }
+
+    companion object {
+        @JvmField
+        val DEFAULT = LanguageVersionSettingsImpl(LanguageVersion.LATEST_STABLE, ApiVersion.LATEST_STABLE)
+    }
+}
```

Comment:
This class already exists in `org.jetbrains.kotlin.config`:

https://github.com/JetBrains/kotlin/blob/c78dbaf6b3290645b1e28fc310b36b3afeca8073/compiler/util/src/org/jetbrains/kotlin/config/LanguageVersionSettings.kt#L273-L317

Also, there might be licensing issues (Kotlin is Apache-licensed, this project is MIT). Do we really want to include this?

Reply:
Removed the class, and referencing internally. For some reason I thought it was private.